src/KubernetesClient/Watch.cs

```diff
@@ -0,0 +1,482 @@
+    public sealed class WatchEvent<T>
+    {
+        /// <summary>Gets or sets the <see cref="WatchEventType"/>.</summary>
+        public WatchEventType Type { get; set; }
+
+        /// <summary>Gets or sets the error that occurred, if the <see cref="Type"/> is <see cref="WatchEventType.Error"/>.</summary>
+        public V1Status Error { get; set; }
+
+        /// <summary>Gets or sets object that the event applies to, if the <see cref="Type"/> is not <see cref="WatchEventType.Error"/>.</summary>
+        public T Object { get; set; }
+    }
+    #endregion
+
+    #region WatchReader
+    /// <summary>Represents a stream of watch events.</summary>
+    /// <typeparam name="T">The type of items to watch</typeparam>
+    public sealed class WatchReader<T> : IDisposable
+    {
+        /// <summary>Initializes a new <see cref="WatchReader{T}"/> from a <see cref="KubernetesResponse"/> containing a watch stream.</summary>
+        public WatchReader(KubernetesResponse response) : this((response ?? throw new ArgumentNullException(nameof(response))).Message) { }
+
+        /// <summary>Initializes a new <see cref="WatchReader{T}"/> from a <see cref="HttpResponseMessage"/> containing a watch stream.</summary>
+        public WatchReader(HttpResponseMessage response)
+        {
+            Response = response ?? throw new ArgumentNullException(nameof(response));
+            if (response.Content == null && !IsError) throw new ArgumentException("The response has no content.");
+        }
+
+        /// <summary>Initializes a new <see cref="WatchReader{T}"/> from a <see cref="Stream"/> containing a watch stream.</summary>
+        public WatchReader(Stream stream) => CreateReader(stream ?? throw new ArgumentNullException(nameof(stream)));
+
+        /// <summary>Indicates whether the request failed and so the <see cref="Response"/> does not contain a valid watch stream.</summary>
+        public bool IsError => Response != null && (int)Response.StatusCode >= 400;
+
+        /// <summary>Gets the underlying <see cref="HttpResponseMessage"/>, or null if the watch reader was initialized from a <see cref="Stream"/>.</summary>
+        public HttpResponseMessage Response { get; }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            streamReader?.Dispose();
+            Response?.Dispose();
+        }
+
+        /// <summary>Reads the next <see cref="WatchEvent{T}"/> from the stream, or returns null if the stream was closed or
+        /// <see cref="IsError"/> is true.
+        /// </summary>
+        public async Task<WatchEvent<T>> ReadAsync(CancellationToken cancelToken = default)
+        {
+            if (IsError) return null;
+            cancelToken.ThrowIfCancellationRequested();
+            if (reader == null) CreateReader(await Response.Content.ReadAsStreamAsync().ConfigureAwait(false));
+            try
+            {
+                streamReader.CancelToken = cancelToken;
+#if NETCOREAPP2_1
```

Comment:
I think this should be `#if !NET452` it's only `net452` that uses v9.0.1.

Reply:
Changed.